PWGGA/GammaConvBase/AliConversionPhotonCuts.cxx

```diff
@@ -4779,6 +4779,63 @@ Bool_t AliConversionPhotonCuts::RejectToCloseV0s(AliAODConversionPhoton* photon,
   return kTRUE;
 }
 
+///________________________________________________________________________
+Bool_t AliConversionPhotonCuts::AllowedBySharedElectronCut(set<Int_t> &theLabels, AliAODConversionPhoton &thePhoton) const {
+  // todo: check if two separate sets for electrons and positrons are safe to use. I think it is and would bring speedup
+
+  // since it is quite rare that this cut cuts, we insert the positive track before checking if the negative one can be inserted as well
+  auto lItInsWorkedPositive = theLabels.insert(thePhoton.GetTrackLabelPositive());
+  if (!lItInsWorkedPositive.second){
+    return  kFALSE;
+  }
+
+  Bool_t lInsWorkedNegative = theLabels.insert(thePhoton.GetTrackLabelNegative()).second;
+  if (!lInsWorkedNegative){
+    theLabels.erase(lItInsWorkedPositive.first);
+    return kFALSE;
+  }
+  return kTRUE;
+}
+
+/* Strategy: keep a list of accepted photons. On inserting a new one, check if it is too close to
+ * any of the ones already on the list
+ * I expect slight changes in comparison to before: other photons will get kicked out and less will get kicked out*/
+Bool_t AliConversionPhotonCuts::AllowedByTooCloseV0sCut(TList &theNotTooClosePhotons, AliAODConversionPhoton &thePhoton) const {
+
+    if (fDoDoubleCountingCut && thePhoton.GetConversionRadius() < fMinRDC){
+      return kTRUE;
+    }
+
+    for (TObject* iObj : theNotTooClosePhotons){
+      AliAODConversionPhoton* iPhoton = dynamic_cast<AliAODConversionPhoton*>(iObj);
+      if (!iPhoton) continue;
+
+      if (!fDoDoubleCountingCut){
+        TVector3 v1(thePhoton.GetConversionX(),
+                    thePhoton.GetConversionY(),
+                    thePhoton.GetConversionZ());
+
+        TVector3 v2(iPhoton->GetConversionX(),
+                    iPhoton->GetConversionY(),
+                    iPhoton->GetConversionZ());
+
+        TVector3 d = v2 - v1;
+
+        if(d.Mag2() < fminV0Dist*fminV0Dist){
+          if(thePhoton.GetChi2perNDF() > iPhoton->GetChi2perNDF()) return kFALSE;
```

Comment:
there still is a bug, if the first photon to be included has a higher chi2 than a following candidate - then the first photon needs to be removed which is still not implemented.

please do not merge until fixed